### PR TITLE
refactor: Remove user-facing parts of STUN-only feature

### LINKED ISSUE #3353: Refactor "STUN-only" feature
After much testing and discussion with @thomaseizinger, we concluded if one client is behind a symmetric NAT, the connection must fallback to TURN.

Because of the likelihood of this happening for our target customer base (roaming clients), it's not reasonable to have a feature in the product that relies on holepunching to succeed. Instead:

- [ ] Remove "Direct routing" vs "Firezone-managed Relays" feature completely from product, marketing, and documentation (sorry @bmanifold)
- [ ] Add ability for Gateway to listen on a specific port. If specified, ensure that peer traffic always uses this port. This guarantees that our existing holepunching implementation will work by using the existing NAT mapping
- [ ] Document the above as "for direct connectivity, open a port..."
- [ ] Ensure a fair bandwidth policy is applied depending on customer tier. The easiest way to achieve this is probably deploy two kinds of Relays: _premium_ and _regular_. When clients/gateways connect, they're given the appropriate list. Changing your billing plan would then involve invalidating Relay credentials, requiring clients to fetch them again (#2040)

### PR DESCRIPTION
**Pull request by jamilbk**
Fixes #3353

### CODE REVIEW

File: elixir/apps/web/lib/web/live/sites/edit.ex [hunk 1]
```diff
@@ -32,65 +31,9 @@ defmodule Web.Sites.Edit do
       <:content>
         <div class="py-8 px-4 mx-auto max-w-2xl lg:py-16">
           <.form for={@form} phx-change={:change} phx-submit={:submit}>
+            <.input type="hidden" name="group[routing]" value="managed" />
```
Comment (bmanifold) [inline elixir/apps/web/lib/web/live/sites/edit.ex]: What do you think of making this:
```
<.input type="hidden" field={@form[:routing]} value="managed" />
```

Was just thinking it follows along with `name` field and would also prevent any issues if we ever needed to change this from `group` to something else in the future.

File: elixir/apps/web/lib/web/live/sites/new.ex [hunk 2]
```diff
@@ -24,61 +23,9 @@ defmodule Web.Sites.New do
       <:content>
         <div class="py-8 px-4 mx-auto max-w-2xl lg:py-16">
           <.form for={@form} phx-change={:change} phx-submit={:submit}>
+            <.input type="hidden" name="group[routing]" value="managed" />
```
Comment (bmanifold) [inline elixir/apps/web/lib/web/live/sites/new.ex]: Same question as above here.

**Review by bmanifold — COMMENTED** [review]
Overall, looks good, just one small suggestion.

**Review by bmanifold — APPROVED** [review]

**Comment by thomaseizinger** [comment]
> **The latest updates on your projects**. Learn more about [Vercel for Git ↗︎](https://vercel.link/github-learn-more)
> 1 Ignored Deployment
> Name 	Status 	Preview 	Comments 	Updated (UTC)
> **firezone** 	⬜️ Ignored ([Inspect](https://vercel.com/firezone/firezone/778TimUpEH489RCcUMS22AzBvQE3)) 	[Visit Preview](https://firezone-git-fix-remove-stun-only-firezone.vercel.app) 		Jan 31, 2024 10:57pm

This doesn't include the portal, does it? It would be cool if it would!

**Pull request merged**